src/encoded/tests/helpers/test_upload_credentials.py

```diff
@@ -0,0 +1,379 @@
+# pylint: disable=protected-access
+"""
+Tests for upload_credential.py
+"""
+from unittest import TestCase
+
+import pytest  # pylint: disable=import-error, unused-import
+
+from moto import mock_sts  # pylint: disable=import-error, unused-import
+
+from encoded.helpers import UploadCredentials
+from encoded.helpers import EXTERNAL_BUCKET_STATEMENTS
+
+from . import HELPERS_DATA_PATH
+
+
+@mock_sts
+def test_external_creds():
+    '''
+    Original test from file.py for test_external_creds
+
+    I do not think the mock works here.
+    '''
+    upload_creds = UploadCredentials(
+        'mock_bucket',
+        'mock_object',
+        'mock_name',
+    )
+    creds = upload_creds.external_creds()
+    assert 'upload_credentials' in creds
+    assert creds['bucket'] == 'mock_bucket'
+    assert creds['key'] == 'mock_object'
+    assert 'mock_name' in creds['upload_credentials']['federated_user_id']
+    assert creds['service'] == 's3'
+
+
+class TestUploadCredentials(TestCase):
```

Comment:
Seems like you could do this in pytest too?

Reply:
Yea, I'll look in to that.  pytest is based on unittest so it may have similar things to use instead.

Reply:
Current Pyramid tutorials uses unittest.  We don't need to import pytest in this file since we're not using @pytest.fixture  

https://docs.pylonsproject.org/projects/pyramid/en/latest/quick_tutorial/more_view_classes.html